django/library/serializers.py

```diff
@@ -62,30 +65,30 @@ class Meta:
 class ContributorSerializer(serializers.ModelSerializer):
     # Need an ID for Vue-Multiselect
     id = serializers.IntegerField(read_only=True)
-    user = RelatedMemberProfileSerializer(required=False, allow_null=True)
+    user = RelatedUserSerializer(required=False, allow_null=True)
     affiliations = TagSerializer(many=True)
-    profile_url = serializers.SerializerMethodField()
+    profile_url = serializers.SerializerMethodField(read_only=True)
 
     def get_existing_contributor(self, validated_data):
         user = validated_data.get("user")
-        email = validated_data.get("email")
         username = user.get("username") if user else None
-        if username is not None:
-            return (
-                User.objects.get(username=username),
-                Contributor.objects.filter(user__username=username).first(),
-            )
+        email = validated_data.get("email")
+        contributor = None
+        # attempt to find contributor by username, then email, then name without an email
+        if username:
+            user = User.objects.filter(username=username).first()
+            contributor = Contributor.objects.filter(user__username=username).first()
         elif email:
-            # match by email if given
             contributor = Contributor.objects.filter(email=email).first()
-            return (contributor.user if contributor else None, contributor)
+            user = contributor.user if contributor else None
         else:
-            # otherwise, match by the exact name given and blank email
             contrib_filter = {
                 k: validated_data.get(k, "")
                 for k in ["given_name", "family_name", "email"]
             }
-            return None, Contributor.objects.filter(**contrib_filter).first()
+            contributor = Contributor.objects.filter(**contrib_filter).first()
```

Comment:
does this clause also need to set `user` to None or is it OK for it to be the result of `validated_data.get("user") at the top of the method`?

Reply:
It may work, but you're right, much better to be explicit